beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestWithInvoiceOptimization.java

```diff
@@ -789,6 +790,181 @@ public void testBillRunInArrearLeadingProration() throws Exception {
+        //
+        // To make it simpler we keep both cuttoffDt on the same value, but we could set any value, e.g P1m, P2m, ...
+        // So, for bill run= Feb 1st -> invoice cuttOffDt = Jan 1st
+        invoiceConfig.setMaxInvoiceLimit(new Period("P0m"));
+
+        clock.setTime(new DateTime("2021-01-15T3:56:02"));
+
+        final Account account = createAccountWithNonOsgiPaymentMethod(getAccountData(15));
+        assertNotNull(account);
+
+        // Don't allow system generated invoices
+        add_AUTO_INVOICING_OFF_Tag(account.getId(), ObjectType.ACCOUNT);
+        // Reuse invoice until committed
+        add_AUTO_INVOICING_DRAFT_Tag(account.getId(), ObjectType.ACCOUNT);
+        add_AUTO_INVOICING_REUSE_DRAFT_Tag(account.getId(), ObjectType.ACCOUNT);
+
+
+        // Create an in-arrear RECURRING subscription
+        final LocalDate effDt1 = new LocalDate(2020, 8, 15);
+        busHandler.pushExpectedEvents(NextEvent.CREATE, NextEvent.BLOCK, NextEvent.BCD_CHANGE);
+        final PlanPhaseSpecifier spec1 = new PlanPhaseSpecifier("blowdart-in-arrear-monthly-notrial");
+        final UUID entitlementId1 = entitlementApi.createBaseEntitlement(account.getId(), new DefaultEntitlementSpecifier(spec1, 15, null, null), null, effDt1, effDt1, false, true, ImmutableList.<PluginProperty>of(), callContext);
+        final Subscription sub1 = subscriptionApi.getSubscriptionForEntitlementId(entitlementId1, callContext);
+        assertListenerStatus();
+
+        // Create an in-arrear USAGE subscription
+        final LocalDate effDt2 = new LocalDate(2020, 9, 1);
+        busHandler.pushExpectedEvents(NextEvent.CREATE, NextEvent.BLOCK, NextEvent.BCD_CHANGE);
+        final PlanPhaseSpecifier spec2 = new PlanPhaseSpecifier("training-usage-in-arrear");
+        final UUID entitlementId2 = entitlementApi.createBaseEntitlement(account.getId(), new DefaultEntitlementSpecifier(spec2, 1, null, null), null, effDt2, effDt2, false, true, ImmutableList.<PluginProperty>of(), callContext);
+        final Subscription sub2 = subscriptionApi.getSubscriptionForEntitlementId(entitlementId2, callContext);
+        assertListenerStatus();
+
+        // Generate some usage data for past months
+        recordUsageData(sub2.getId(), "tracking-old-1", "hours", new LocalDate(2020, 9, 19), 1L, callContext);
+        recordUsageData(sub2.getId(), "tracking-old-2", "hours", new LocalDate(2020, 10, 19), 1L, callContext);
+        recordUsageData(sub2.getId(), "tracking-old-3", "hours", new LocalDate(2020, 11, 19), 1L, callContext);
+        recordUsageData(sub2.getId(), "tracking-old-4", "hours", new LocalDate(2020, 12, 19), 1L, callContext);
+
+        // Generate usage data for this month
+        recordUsageData(sub2.getId(), "tracking-1", "hours", new LocalDate(2021, 1, 19), 1L, callContext);
+
+        // 2021-02-01
+        // (Hum... Interestingly, there is no future notification set on the 1st although sub2#BCD is the 1st...)
```

Comment:
Were you expecting a notification for an invoice run?

When we process the initial subscription events, we return early because of `AUTO_INVOICING_OFF` https://github.com/killbill/killbill/blob/3a172831303caccdb0e16d0091d660a97089d8fa/invoice/src/main/java/org/killbill/billing/invoice/InvoiceDispatcher.java#L367

Reply:
I would expect to see an entry in the notificationQ for `2021-02-01`, because there is something to invoice for that date (the usage subscription is aligned on the 1st). However the notification does not exist (-> I think this is a bug). However, because of `AUTO_INVOICING_OFF` I do not expect a `NULL_INVOICE` (as you said)